src/main/java/org/reactome/release/qa/common/QACheckerHelper.java

```diff
@@ -225,5 +232,339 @@ public static String getAttributeTableName(String clsName,
 	    else
 	        return originCls.getName();
 	}
-	
+
+	/**
+	 * Finds all Events in DB that are not used for manual inference, or members of Pathways in skiplist. Finding if
+	 * manually inferred is done by checking for a null 'inferredFrom' referral.
+	 * @param dba MySQLAdaptor
+	 * @param skiplistDbIds List<String> -- List of Pathway DbIds. If an Event being checked is a member of these pathways, they are skipped.
+	 * @return Set<GKInstance> -- All Events in DB that are not used for manual inference.
+	 * @throws Exception -- Thrown by MySQLAdaptor
+	 */
+	public static Set<GKInstance> findEventsNotUsedForManualInference(MySQLAdaptor dba, List<String> skiplistDbIds) throws Exception {
+		Set<GKInstance> eventsNotUsedForInference = new HashSet<>();
+		Collection<GKInstance> events = dba.fetchInstancesByClass(ReactomeJavaConstants.Event);
+		for (GKInstance event : events) {
+			if (!manuallyInferred(event) && !memberSkipListPathway(event, skiplistDbIds)) {
+				eventsNotUsedForInference.add(event);
+			}
+		}
+		return eventsNotUsedForInference;
+	}
+
+	/**
+	 * Finds all Events not used for Inference, and then finds subset that are Human ReactionlikeEvents
+	 * @param dba MySQLAdaptor
+	 * @param skiplistDbIds List<String> -- List of Pathway DbIds. If an Event being checked is a member of these pathways, they are skipped.
+	 * @return Set<GKInstance> -- All Human ReactionlikeEvents that are not used for manual inference.
+	 * @throws Exception-- Thrown by MySQLAdaptor
+	 */
+	public static Set<GKInstance> findHumanReactionsNotUsedForManualInference(MySQLAdaptor dba, List<String> skiplistDbIds) throws Exception {
+		Set<GKInstance> reactionsNotUsedForManualInference = new HashSet<>();
+		for (GKInstance event : findEventsNotUsedForManualInference(dba, skiplistDbIds)) {
+			// Filter for Human ReactionlikeEvents
+			if (event.getSchemClass().isa(ReactomeJavaConstants.ReactionlikeEvent)
+					&& isHumanDatabaseObject(event)) {
+
+				reactionsNotUsedForManualInference.add(event);
+			}
+		}
+		return reactionsNotUsedForManualInference;
+	}
+
+	/**
+	 * Finds all PhysicalEntities that exist in a ReactionlikeEvent's inputs, outputs, catalysts and regulations.
+	 * @param reaction GKInstance -- ReactionlikeEvent that will be searched for PhysicalEntities.
+	 * @return Set<GKInstance> -- All distinct PhysicalEntitys in ReactionlikeEvent
+	 * @throws Exception -- Thrown by MySQLAdaptor
+	 */
+	public static Set<GKInstance> findAllPhysicalEntitiesInReaction(GKInstance reaction) throws Exception {
+		Set<GKInstance> reactionPEs = new HashSet<>();
+		reactionPEs.addAll(findAllInputAndOutputPEs(reaction));
+		reactionPEs.addAll(findAllCatalystPEs(reaction));
+		reactionPEs.addAll(findAllRegulationPEs(reaction));
+		return reactionPEs;
+	}
```

Comment:
Please use org.gk.model.InstanceUtilities.getReactionparticipants(GKInstance) so that we will have the same logic across all codebase.

Reply:
OK, I'll do that. Does this method get all catalyst and regulation participants as well?